beacon-light-client/plonky2/crates/circuit_executables/src/provers.rs

```diff
@@ -27,28 +28,46 @@ use starky_bls12_381::{
 pub fn prove_inner_level(
     proof1_bytes: Vec<u8>,
     proof2_bytes: Vec<u8>,
-    inner_circuit_data: &CircuitData<GoldilocksField, PoseidonGoldilocksConfig, 2>,
-    inner_circuit_target: &BasicRecursiveInnerCircuitTarget,
     circuit_data: &CircuitData<GoldilocksField, PoseidonGoldilocksConfig, 2>,
+    inner_circuit_target: &BasicRecursiveInnerCircuitTarget,
+    inner_circuit_data: &CircuitData<GoldilocksField, PoseidonGoldilocksConfig, 2>,
 ) -> Result<ProofWithPublicInputs<GoldilocksField, PoseidonGoldilocksConfig, 2>> {
     let inner_proof1 =
         ProofWithPublicInputs::<GoldilocksField, PoseidonGoldilocksConfig, 2>::from_bytes(
             proof1_bytes,
-            &inner_circuit_data.common,
+            &circuit_data.common,
         )?;
 
     let inner_proof2 =
         ProofWithPublicInputs::<GoldilocksField, PoseidonGoldilocksConfig, 2>::from_bytes(
             proof2_bytes,
-            &inner_circuit_data.common,
+            &circuit_data.common,
         )?;
 
     let mut pw = PartialWitness::new();
 
     pw.set_proof_with_pis_target(&inner_circuit_target.proof1, &inner_proof1);
     pw.set_proof_with_pis_target(&inner_circuit_target.proof2, &inner_proof2);
 
-    Ok(circuit_data.prove(pw)?)
+    Ok(inner_circuit_data.prove(pw)?)
+}
+
+// TODO: Don't hard code the D
```

Comment:
TODO?

Reply:
I couldn't do it for some reason that I don't remember. It definitely must be done and will be done, just not now. This is a second version of the prove_inner_level function, it's even more constrained, so it's fine. Both of the functions need some updating